src/pages/ModerationPage.tsx

```diff
@@ -25,11 +26,33 @@ export default function ImageModerationPage() {
     const [Tickets, setTickets] = useState<Ticket[]>([])
 
     useEffect(() => {
-        // send get request to api to get tickets and set Tickets to the response
-        axios.get(`${import.meta.env.VITE_API_URL}/tickets?status=open`).then((res) => {
-            setTickets(res.data)
-        })
+        const fetchTickets = async () => {
+            try {
+                const response = await axios.get(`${import.meta.env.VITE_API_URL}/tickets?status=open`);
+                const ticketsData = response.data;
+                setTickets(ticketsData);
+
+                const ticketIds = ticketsData.map((ticket: any) => ticket.id);
+                const flagsResponse = await axios.post(`${import.meta.env.VITE_API_URL}/flags/batch`, {
+                    ticket_ids: ticketIds
+                });
+
+                // update ticket by adding reasons from flags
+                const ticketIdToFlags = flagsResponse.data.ticket_id_to_flags;
+                const updatedTickets = ticketsData.map((ticket: any) => {
+                    ticket.reasons = ticketIdToFlags[ticket.id].map((flag: any) => flag.reason);
+                    return ticket;
+                });
+
+                setTickets(updatedTickets);
+            } catch (err) {
+                console.error(err)
+            }
+        };
+        fetchTickets()
```

Comment:
Does the way I solved the problem seem correct to you?

Reply:
Yes it looks good to me!